### PR TITLE
SSCS-7748 Handle missing representative names in cover letters

### PR DESCRIPTION
**Pull request by danhamilton**
**Before creating a pull request make sure that:**

- [ X] commit messages are meaningful and follow good commit message guidelines
- [ X] README and other documentation has been updated / added (if needed)
- [ X] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-7478

### Change description ###
Handle missing representative names in cover letters to fall back to organisation name or Sir/Madam

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -268,6 +268,8 @@ dependencyManagement {
       entry 'jackson-databind'
     }
 
+    dependency group: 'org.apache.tika', name: 'tika-core', version: '1.24.1'
```
Comment (ryandac) [inline build.gradle:271]: Is this needed?

File: src/main/java/uk/gov/hmcts/reform/sscs/service/MockBulkPrintService.java [hunk 2]
```diff
@@ -18,7 +18,10 @@
     public Optional<UUID> sendToBulkPrint(List<Pdf> pdfs, SscsCaseData sscsCaseData) {
 
         logger.info("No bulk print operation needs to be performed as 'Bulk print url' is switched off.");
-
         return Optional.of(UUID.fromString("abc123ca-c336-11e9-9cb5-123456789abc"));
     }
+
+
+
+
```
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/MockBulkPrintService.java:26]: lots of unnecessary blank lines in this change

File: src/main/java/uk/gov/hmcts/reform/sscs/service/placeholders/FurtherEvidencePlaceholderService.java [hunk 3]
```diff
@@ -26,38 +26,77 @@ public FurtherEvidencePlaceholderService(PlaceholderService placeholderService)
...
+                .map(Representative::getOrganisation)
+                .filter(org -> isNonEmptyNotNull(org))
+                .orElse("Sir/Madam"));
+    }
+
+    private Boolean isValidName(Name name) {
+        return (isNonEmptyNotNull(name.getFirstName()) && isNonEmptyNotNull(name.getLastName())) ? true : false;
+    }
+
+    private Boolean isNonEmptyNotNull(String field) {
```
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/placeholders/FurtherEvidencePlaceholderService.java:96]: Consider using https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#isNotBlank-java.lang.CharSequence-

File: src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/ReciprocalLinkHandler.java [hunk 4]
```diff
@@ -82,12 +82,17 @@ public void handle(CallbackType callbackType, Callback<SscsCaseData> callback) {
     }
 
     private void backLinkAssociatedCases(Long caseId, List<SscsCaseDetails> matchedByNinoCases, IdamTokens idamTokens) {
+
```
Comment (srochani) [inline src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/ReciprocalLinkHandler.java:85]: No changes in file, can we please revert this blank line

File: src/main/java/uk/gov/hmcts/reform/sscs/service/CoverLetterService.java [hunk 5]
```diff
@@ -56,12 +56,17 @@ private void printCoverLetterToPdfLocallyForDebuggingPurpose(byte[] coverLetterC
...
+
         requireNonNull(caseData, "caseData must not be null");
+
+
         byte[] coverLetterContent = pdfGenerationService.generatePdf(DocumentHolder.builder()
             .template(new Template(templateName, hmctsDocName))
             .placeholders(furtherEvidencePlaceholderService.populatePlaceholders(caseData, letterType))
             .pdfArchiveMode(true)
             .build());
+
```
Comment (srochani) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/CoverLetterService.java:68]: No changes in file, can we please revert this blank line

File: src/main/java/uk/gov/hmcts/reform/sscs/service/MockBulkPrintService.java [hunk 6]
```diff
@@ -18,7 +18,10 @@
     public Optional<UUID> sendToBulkPrint(List<Pdf> pdfs, SscsCaseData sscsCaseData) {
 
         logger.info("No bulk print operation needs to be performed as 'Bulk print url' is switched off.");
-
         return Optional.of(UUID.fromString("abc123ca-c336-11e9-9cb5-123456789abc"));
     }
+
```
Comment (srochani) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/MockBulkPrintService.java:23]: No changes in file, can we please revert this blank line

File: src/main/java/uk/gov/hmcts/reform/sscs/service/FurtherEvidenceService.java [hunk 7]
```diff
@@ -54,8 +54,11 @@ public void issue(List<SscsDocument> sscsDocuments, SscsCaseData caseData, Docum
 
     private void send609_97_OriginalSender(SscsCaseData caseData, DocumentType documentType, List<Pdf> pdfs,
                                            List<FurtherEvidenceLetterType> allowedLetterTypes) {
+
+
```
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/FurtherEvidenceService.java:58]: there are blank line changes in this file

File: src/main/java/uk/gov/hmcts/reform/sscs/service/CoverLetterService.java [hunk 8]
```diff
@@ -56,12 +56,17 @@ private void printCoverLetterToPdfLocallyForDebuggingPurpose(byte[] coverLetterC
 
     public byte[] generateCoverLetter(SscsCaseData caseData, FurtherEvidenceLetterType letterType, String templateName,
                                       String hmctsDocName) {
+
         requireNonNull(caseData, "caseData must not be null");
+
+
```
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/CoverLetterService.java:62]: there are no changes to this file

File: src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/ReciprocalLinkHandler.java [hunk 9]
```diff
@@ -82,12 +82,17 @@ public void handle(CallbackType callbackType, Callback<SscsCaseData> callback) {
...
+
```
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/ReciprocalLinkHandler.java:86]: there are no changes here. Ignore the comment if the formatting is wrong.

File: build.gradle (hunk 1 above)
Comment (danhamilton) [reply build.gradle:271]: yes - solves OWASP CVE

**Pull request merged**